src/ucs/sys/sys.c

```diff
@@ -767,23 +767,52 @@ ucs_status_t ucs_sys_get_proc_cap(uint32_t *effective)
 #endif
 }
 
-static void ucs_sysv_shmget_error_check_EPERM(int flags, char *buf, size_t max)
+/**
+ * Checks whether the process have memory lock capability.
+ *
+ * @param [out] mlock_cap Pointer to a flag that indicates whether the
+ *              process have the required capability.
+ *
+ * @return UCS_OK or error in case of failure.
+ */
+
+static ucs_status_t ucs_sys_get_mlock_cap(int *mlock_cap)
```

Comment:
why not return 0/1 ?
seems in all places a failure is treated same way as mlock_cap==0

Reply:
Status is used in `ucs_sysv_shmget_error_check_EPERM`.

Reply:
i think the slightly different message does not justify the extra "confusion"
ucs_sys_check_mlock_cap() returns 1 if have memlock capability, or 0 if do not have the capability or it could not be detected.